examples/transform/src/main/java/org/apache/apex/examples/transform/Application.java

```diff
@@ -19,25 +19,72 @@
 
 package org.apache.apex.examples.transform;
 
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
 import org.apache.hadoop.conf.Configuration;
 
 import com.datatorrent.api.Context;
 import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
 import com.datatorrent.api.StreamingApplication;
 import com.datatorrent.api.annotation.ApplicationAnnotation;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+
 import com.datatorrent.common.partitioner.StatelessPartitioner;
+import com.datatorrent.common.util.BaseOperator;
 import com.datatorrent.lib.io.ConsoleOutputOperator;
 import com.datatorrent.lib.transform.TransformOperator;
 
-@ApplicationAnnotation(name = "TransformExample")
 /**
  * @since 3.7.0
  */
+@ApplicationAnnotation(name = "TransformExample")
 public class Application implements StreamingApplication
 {
+  public static class Collector extends BaseOperator
```

Comment:
Instead of adding all of this, I have an alternative proprosal:

- Use the FunctionOperator instead of Console...
- Add member outputFunction to the application class (default console)
- Override the outputFunction in the test to collect in test private static collection.

That will keep the code more concise yet allow proper testing.